src/qibojit/tests/test_gates.py

```diff
@@ -135,3 +135,27 @@ def test_apply_fsim(backend, nqubits, targets, controls, dtype):
     state = op.apply_fsim(state, gate, nqubits, target1, target2, qubits)
     state = op.to_numpy(state)
     np.testing.assert_allclose(state, target_state, atol=ATOL.get(dtype))
+
+
+@pytest.mark.parametrize(("nqubits", "targets", "controls"),
+                         [(3, [0, 1, 2], []), (4, [2, 1, 3], []),
+                          (5, [0, 2, 3], []), (8, [2, 6, 3], []),
+                          (5, [0, 2, 3, 4], []), (8, [0, 4, 2, 5, 7], []),
+                          (7, [0, 2, 4, 3, 6, 5], []), (8, [0, 4, 2, 3, 5, 7, 1], []),
+                          (4, [2, 1, 3], [0]), (5, [0, 2, 3], [1]),
+                          (8, [2, 6, 3], [4, 7]), (5, [0, 2, 3, 4], [1]),
+                          (8, [0, 4, 2, 5, 7], [1, 3]),
+                          (10, [0, 4, 2, 5, 9], [1, 3, 7, 8])])
+def test_apply_multiqubit_gate(nqubits, targets, controls, dtype):
```

Comment:
@stavros11 I have a question here, does this test runs on GPU, I mean if we include a `backend` argument, it is always None, but I don't see in conf the setup to exclude cupy. Could you please remind me how this parsing works in qibojit?

Reply:
I believe the qibojit tests just use the default device. If you see the [qibojit backend initialization](https://github.com/qiboteam/qibojit/blob/5d55240cb3604cb4d1d2a885b226df0e864e1b1a/src/qibojit/custom_operators/__init__.py#L9), it will use cupy if that is available otherwise it will fall back to numba. So if your machine has GPU it will test cupy which is the default, otherwise (or if you hide the GPU with `export CUDA_VISIBLE_DEVICES=""`) it will fall back to numba and test CPU only. CI tests CPU as there is no GPU.

For the multiqubit kernel in particular, I have it [fall back to CPU](https://github.com/qiboteam/qibojit/blob/db605a560d7422147ef759a6b6fe2c3baaadd684/src/qibojit/custom_operators/__init__.py#L89) temporarily until we implement it on GPU. This will be fixed in #23. So if you try the tests on this branch they should pass regardless of device. However if you run them on #23 using GPU you can see that they will fail if you change something in the GPU kernel.

Reply:
Ok, thanks, it is clear.